addon/-private/system/relationships/state/has-many.js

```diff
@@ -209,7 +209,7 @@ export default class ManyRelationship extends Relationship {
 
     for (let i = 0; i< internalModels.length; i++) {
       let internalModel = internalModels[i];
-      if (this.canonicalMembers.has(internalModel)) {
```

Comment:
@stefanpenner I wondered that too, and I think could be a side effect here: https://github.com/emberjs/data/blob/master/addon/-private/system/relationships/state/relationship.js#L167 which add internal models to the relationship canoninal members. So this may leads to `this.canonicalMembers` and `forCanonical`beeing different